src/Microsoft.Extensions.DependencyModel/DependencyContextJsonReader.cs

```diff
@@ -28,23 +28,92 @@ public DependencyContext Read(Stream stream)
+
+                if (!string.IsNullOrEmpty(runtimeTargetInfo.Name))
+                {
+                    runtimeTarget = (JObject) targetsObject.FirstOrDefault(t => t.Key == runtimeTargetInfo.Name).Value;
+                    if (runtimeTarget == null)
+                    {
+                        throw new FormatException($"Target with name {runtimeTargetInfo.Name} not found");
+                    }
+
+                    var seperatorIndex = runtimeTargetInfo.Name.IndexOf(DependencyContextStrings.VersionSeperator);
+                    if (seperatorIndex > -1 && seperatorIndex < runtimeTargetInfo.Name.Length)
+                    {
+                        runtime = runtimeTargetInfo.Name.Substring(seperatorIndex + 1);
+                    }
+                }
+                else
+                {
+                    runtimeTarget = compileTarget;
+                }
+            }
 
             return new DependencyContext(
-                compileTargetProperty.Key,
-                runtimeTargetProperty.Key.Substring(compileTargetProperty.Key.Length + 1),
+                target,
+                runtime,
+                runtimeTargetInfo.Portable,
                 ReadCompilationOptions((JObject)root[DependencyContextStrings.CompilationOptionsPropertName]),
-                ReadLibraries((JObject)compileTargetProperty.Value, false, libraryStubs).Cast<CompilationLibrary>().ToArray(),
-                ReadLibraries((JObject)runtimeTargetProperty.Value, true, libraryStubs).Cast<RuntimeLibrary>().ToArray()
+                ReadLibraries(compileTarget, false, libraryStubs).Cast<CompilationLibrary>().ToArray(),
+                ReadLibraries(runtimeTarget, true, libraryStubs).Cast<RuntimeLibrary>().ToArray(),
+                ReadRuntimeGraph((JObject)root[DependencyContextStrings.RuntimesPropertyName]).ToArray()
                 );
         }
 
+        private IEnumerable<KeyValuePair<string, string[]>> ReadRuntimeGraph(JObject runtimes)
+        {
+            if (runtimes == null)
+            {
+                yield break;
+            }
+
+            var targets = runtimes.Children();
+            var runtime = (JProperty)targets.Single();
+            foreach (var pair in (JObject)runtime.Value)
+            {
+                yield return new KeyValuePair<string, string[]>(pair.Key, pair.Value.Values<string>().ToArray());
+            }
+        }
+
+        private RuntimeTargetInfo ReadRuntimeTargetInfo(JObject root)
+        {
+
+            var runtimeTarget = (JObject)root[DependencyContextStrings.RuntimeTargetPropertyName];
+            if (runtimeTarget != null)
+            {
+                return new RuntimeTargetInfo()
+                {
+                    Name = runtimeTarget[DependencyContextStrings.RuntimeTargetNamePropertyName]?.Value<string>(),
+                    Portable = runtimeTarget[DependencyContextStrings.PortablePropertyName]?.Value<bool>() == true
```

Comment:
This is now just a simple string property, see https://github.com/dotnet/cli/blob/anurse/update-config-spec/Documentation/specs/runtime-configuration-file.md#appnameruntimeconfigjson

That changed recently :). The host will use the presence of the `runtimeOptions.framework` section in the `.runtimeconfig.json` file to determine if the target is portable.